src/main/java/com/project/controller/PasswordResetController.java

```diff
@@ -0,0 +1,65 @@
+package com.project.controller;
+
+import com.project.exception.UserNotFoundException;
+import com.project.model.User;
+import com.project.model.request.PasswordResetRequest;
+import com.project.model.response.GenericResponse;
+import com.project.service.EmailService;
+import com.project.service.PasswordResetService;
+import com.project.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.UUID;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("api/users")
+public class PasswordResetController {
+
+    private final UserService userService;
+    private final EmailService emailService;
+    private final PasswordResetService passwordResetService;
+
+    @Autowired
+    public PasswordResetController(UserService userService, EmailService emailService, PasswordResetService passwordResetService){
+        this.userService = userService;
+        this.emailService = emailService;
+        this.passwordResetService = passwordResetService;
+    }
+
+    @GetMapping("/changePassword")
+    public String showChangePasswordPage(Model model, @RequestParam("id") long id, @RequestParam("token") String token) {
```

Comment:
It looks to me that this service is more for validating the password reset token rather than showing some page. I would suggest returning just an empty response if token was validated successfully and throw an error otherwise.

Reply:
this service validates the token and, depending on whether its valid or not, returns the path to which the user should be redirected.